### PR TITLE
[keybindings] add 'Reset' button to the edit keybinding dialog

### PR DESCRIPTION
**Pull request by vince-fugnitto**
**Description**

- created a custom `EditKeybinding` dialog used to edit keybindings, and reset custom
keybindings to their default value. Custom keybindings are keybindings which have
a `User` scope (exist in the `keymaps.json`).
- the additional functionality added to the edit keybindings dialog is
used to more easily revert custom keybindings without the need to click
the revert button in the tree.

---

<div align='center'>

Screen Shot 2019-06-27 at 10 00 06 PM

</div>

**How to Test**
1. open the keybindings-widget: <kbd>F1</kbd> then select `Open Keyboard Shortcuts`.
2. update a command's keybinding (using the edit button or double click the row in the table).
3. click the edit again (once a custom keybinding has been set).
4. notice the `Reset` button.
5. click the `Reset` button.
6. the keybinding for the command should be properly reset to it's default value.

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by lmcbout** [comment]
While testing, I added a filter in the "Search keybindings" field and no "Reset" button was available to return to the default value. As a filter I entered "user" to filter the modified keybindings.

**Comment by vince-fugnitto** [comment]
> While testing, I added a filter in the "Search keybindings" field and no "Reset" button was available to return to the default value. As a filter I entered "user" to filter the modified keybindings.

Do you have any command with a custom keybinding?
Only commands which are edited will display the `reset` in the table, and the `reset` in the dialog.

**Comment by fangnx** [comment]
When I tested, it seemed I could still use the old/default keybinding after the edit.

To reproduce this:
1. Change the keybinding for `Add Line Comment` from `ctrl+k ctrl+x` to something else (`ctrl+k ctrl+p`, for example).
2. Go to the editor. Both the old and new keybindings should work.

**Comment by lmcbout** [comment]
> > While testing, I added a filter in the "Search keybindings" field and no "Reset" button was available to return to the default value. As a filter I entered "user" to filter the modified keybindings.
> 
> @lmcbout
> 
> Do you have any command with a custom keybinding?
> Only commands which are edited will display the `reset` in the table, and the `reset` in the dialog.

@vince-fugnitto 
I have a few custom keybindings and the reset is not showing

**Comment by vince-fugnitto** [comment]
> > > While testing, I added a filter in the "Search keybindings" field and no "Reset" button was available to return to the default value. As a filter I entered "user" to filter the modified keybindings.
> > 
> > 
> > @lmcbout
> > Do you have any command with a custom keybinding?
> > Only commands which are edited will display the `reset` in the table, and the `reset` in the dialog.
> 
> @vince-fugnitto
> I have a few custom keybindings and the reset is not showing

Could you show me? A gif perhaps.

**Comment by lmcbout** [comment]
Also, the reset on the keybinding shortcut page disseapear when entering a filter. See the video, The "Add Folder to workspace" has the rest option until I add a filter, then it only dislay the"Edit"icon"

KeyBindingReset

**Comment by vince-fugnitto** [comment]
@lmcbout thank you Jacques, the `reset` button and icon are now present if the search matches the `scope` column. You can kindly re-test whenever you get the chance.

File: packages/keymaps/src/browser/keybindings-widget.tsx [hunk 1]
```diff
@@ -631,3 +631,85 @@ export class KeybindingWidget extends ReactWidget {
...
+        const id = this.getRaw(this.item.id);
+        // Remove the custom keybinding, resetting it to its default value.
+        this.keymapsService.removeKeybinding(id);
+    }
+
+    /**
+     * Extract the raw value from a string (without fuzzy matching).
+     * @param a the string value.
+     */
+    protected getRaw(a: string): string {
```
Comment (lmcbout) [inline packages/keymaps/src/browser/keybindings-widget.tsx]: Minor: Some methods returning something, you put it in the header section, but here, you did not put the return in the header comment ! @return ....

**Comment by vince-fugnitto** [comment]
Thanks guys! @lmcbout @fangnx

**Pull request merged**